src/coreclr/jit/utils.cpp

```diff
@@ -2294,6 +2294,22 @@ bool FloatingPointUtils::isNaN(double val)
     return (bits & 0x7FFFFFFFFFFFFFFFULL) > 0x7FF0000000000000ULL;
 }
 
+//------------------------------------------------------------------------
+// isNegativeZero: Determines whether the specified value is negative zero (-0.0)
+//
+// Arguments:
+//    val - value to check for (-0.0)
+//
+// Return Value:
+//    True if val is (-0.0)
+//
+
+bool FloatingPointUtils::isNegativeZero(double val)
+{
+    UINT64 bits = *reinterpret_cast<UINT64*>(&val);
+    return (bits >> 63) == 1;
+}
```

Comment:
This looks wrong, won't it return true for any negative value?

Reply:
Yep, it needs to be explicitly `bits == 0x8000000000000000ULL`

Reply:
Thank you, I missed it :(